### ISSUE TITLE
False positive case of "Not all arguments converted during string formatting"

### ISSUE THREAD

**Issue by jfcherng** [python/mypy#15269]
**Bug Report**

Using `StrEnum`'s `str.format` without via an intermedium varible makes mypy unhappy.

**To Reproduce**

`test.py`:

```py
from enum import StrEnum


class Animal(StrEnum):
    DOG = "dog: {name}"


dog = Animal.DOG
print(dog.format(name="Goffy"))  # mypy happy

print(Animal.DOG.format(name="Goffy"))  # mypy unhappy
```

An alternative snippet for `Python < 3.11`:

```py
from enum import Enum


class Animal(str, Enum):
    DOG = "dog: {name}"


dog = Animal.DOG
print(dog.format(name="Goffy"))  # mypy happy

print(Animal.DOG.format(name="Goffy"))  # mypy unhappy
```

**Expected Behavior**

No error.

**Actual Behavior**

```
test.py:11: error: Not all arguments converted during string formatting  [str-format]
```

**Your Environment**

- Mypy version used: 1.7.0
- Mypy command-line flags: Literally none. (i.e., `$ mypy test.py`)
- Mypy configuration options from `mypy.ini` (and other config files): None.
- Python version used: 3.12.0

**Comment by setu4993** [comment]
I believe this was fixed in https://github.com/python/mypy/pull/16555 since we aren't running into this since updating to v1.10.0.

**Comment by jfcherng** [comment]
> I believe this was fixed in #16555 since we aren't running into this since updating to v1.10.0.

Fixed confirmed. Thank you for informing this ❤️
